Update doctests for numpy 2.0.0 scaler repr changes

See https://numpy.org/neps/nep-0051-scalar-representation.html - for example this:

```pycon
>>> np.float32(3.0)
3.0
```

becomes:

```pycon
>>> np.float32(3.0)
np.float32(3.0)
```

As per the NEP, this is expected to require documentation updates in libraries using numpy - like Biopython. Out of practicality we will probably do this alongside setting a minimum dependency of numpy 2.0.0

@mgalardini do you have any views on how the Biopython phenotype module should handle the numpy scalar repr changes? e.g. Should we just update the doctests for numpy 2 onwards, or are you comfortable changing the code to cast to the Python float?

In addition to the tutorial test failures, running the doctests too there is currently only one file failing, ``Bio/phenotype/phen_micro.py``:

```
======================================================================
FAIL: WellRecord (Bio.phenotype.phen_micro)
Doctest: Bio.phenotype.phen_micro.WellRecord
----------------------------------------------------------------------
Traceback (most recent call last):
  File "/Users/pc40583/mambaforge/lib/python3.10/doctest.py", line 2221, in runTest
    raise self.failureException(self.format_failure(new.getvalue()))
AssertionError: Failed doctest test for Bio.phenotype.phen_micro.WellRecord
  File "/Users/pc40583/repositories/biopython/Bio/phenotype/phen_micro.py", line 580, in WellRecord

----------------------------------------------------------------------
File "/Users/pc40583/repositories/biopython/Bio/phenotype/phen_micro.py", line 597, in Bio.phenotype.phen_micro.WellRecord
Failed example:
    well[1]
Expected:
    16.0
Got:
    np.float64(16.0)
----------------------------------------------------------------------
File "/Users/pc40583/repositories/biopython/Bio/phenotype/phen_micro.py", line 599, in Bio.phenotype.phen_micro.WellRecord
Failed example:
    well[1:5]
Expected:
    [16.0, 20.0, 18.0, 15.0]
Got:
    [np.float64(16.0), np.float64(20.0), np.float64(18.0), np.float64(15.0)]
----------------------------------------------------------------------
File "/Users/pc40583/repositories/biopython/Bio/phenotype/phen_micro.py", line 601, in Bio.phenotype.phen_micro.WellRecord
Failed example:
    well[1:5:0.5]
Expected:
    [16.0, 19.0, 20.0, 18.0, 18.0, 18.0, 15.0, 18.0]
Got:
    [np.float64(16.0), np.float64(19.0), np.float64(20.0), np.float64(18.0), np.float64(18.0), np.float64(18.0), np.float64(15.0), np.float64(18.0)]
----------------------------------------------------------------------
File "/Users/pc40583/repositories/biopython/Bio/phenotype/phen_micro.py", line 608, in Bio.phenotype.phen_micro.WellRecord
Failed example:
    well[1.3]
Expected:
    19.0
Got:
    np.float64(19.0)
----------------------------------------------------------------------
File "/Users/pc40583/repositories/biopython/Bio/phenotype/phen_micro.py", line 610, in Bio.phenotype.phen_micro.WellRecord
Failed example:
    well[1250]
Expected:
    nan
Got:
    np.float64(nan)
```

--- a/Bio/Entrez/__init__.py
+++ b/Bio/Entrez/__init__.py
@@ -358,12 +358,12 @@ def egquery(**keywds):
 
     >>> from Bio import Entrez
     >>> Entrez.email = "Your.Name.Here@example.org"
-    >>> handle = Entrez.egquery(term="biopython")
-    >>> record = Entrez.read(handle)
-    >>> handle.close()
-    >>> for row in record["eGQueryResult"]:
-    ...     if "pmc" in row["DbName"]:
-    ...         print(int(row["Count"]) > 60)
+    >>> handle = Entrez.egquery(term="biopython")  # doctest: +SKIP
+    >>> record = Entrez.read(handle)  # doctest: +SKIP
+    >>> handle.close()  # doctest: +SKIP
+    >>> for row in record["eGQueryResult"]:  # doctest: +SKIP
+    ...     if "pmc" in row["DbName"]:  # doctest: +SKIP
+    ...         print(int(row["Count"]) > 60)  # doctest: +SKIP
     True
 
     :returns: Handle to the results, by default in XML format.

--- a/Bio/motifs/matrix.py
+++ b/Bio/motifs/matrix.py
@@ -32,7 +32,8 @@ class GenericPositionMatrix(dict):
                 self.length = len(values[letter])
             elif self.length != len(values[letter]):
                 raise Exception("data has inconsistent lengths")
-            self[letter] = list(values[letter])
+            # Cast any numpy floats into Python floats:
+            self[letter] = [float(_) for _ in values[letter]]
         self.alphabet = alphabet
 
     def __str__(self):

--- a/Tests/test_PDB_PDBMLParser.py
+++ b/Tests/test_PDB_PDBMLParser.py
@@ -6,9 +6,11 @@ returned by the mmCIF parser for any PDB structure.
 """
 
 import unittest
+import warnings
 
 from Bio.PDB import MMCIFParser
 from Bio.PDB import PDBMLParser
+from Bio.PDB.PDBExceptions import PDBConstructionWarning
 
 
 class TestPDBMLParser(unittest.TestCase):
@@ -16,21 +18,25 @@ class TestPDBMLParser(unittest.TestCase):
         mmcif_parser = MMCIFParser()
         pdbml_parser = PDBMLParser()
 
-        for entry in ["1GBT", "6WG6", "3JQH"]:
-            mmcif_structure = mmcif_parser.get_structure(entry, f"PDB/{entry}.cif")
-            pdbml_structure = pdbml_parser.get_structure(f"PDB/{entry}.xml")
+        with warnings.catch_warnings():
+            warnings.simplefilter("ignore", PDBConstructionWarning)
+            for entry in ["1GBT", "6WG6", "3JQH"]:
+                mmcif_structure = mmcif_parser.get_structure(entry, f"PDB/{entry}.cif")
+                pdbml_structure = pdbml_parser.get_structure(f"PDB/{entry}.xml")
             self.assertEqual(mmcif_structure, pdbml_structure)
 
     def test_get_structure_filehandle(self):
         mmcif_parser = MMCIFParser()
         pdbml_parser = PDBMLParser()
 
-        for entry in ["1GBT"]:
-            with open(f"PDB/{entry}.cif") as mmcif_file, open(
-                f"PDB/{entry}.xml"
-            ) as pdbml_file:
-                mmcif_structure = mmcif_parser.get_structure(entry, mmcif_file)
-                pdbml_structure = pdbml_parser.get_structure(pdbml_file)
+        with warnings.catch_warnings():
+            warnings.simplefilter("ignore", PDBConstructionWarning)
+            for entry in ["1GBT"]:
+                with open(f"PDB/{entry}.cif") as mmcif_file, open(
+                    f"PDB/{entry}.xml"
+                ) as pdbml_file:
+                    mmcif_structure = mmcif_parser.get_structure(entry, mmcif_file)
+                    pdbml_structure = pdbml_parser.get_structure(pdbml_file)
                 self.assertEqual(mmcif_structure, pdbml_structure)
 
 

--- a/Tests/test_SeqFeature.py
+++ b/Tests/test_SeqFeature.py
@@ -367,14 +367,14 @@ class TestExtract(unittest.TestCase):
                 "join{[0:2](-), [3:4](-)}",
             )
 
-        # Origin-spanning location containing the entire sequence
-        self.assertEqual(
-            str(SimpleLocation.fromstring("3..2", 4, True)), "join{[2:4], [0:2]}"
-        )
-        self.assertEqual(
-            str(SimpleLocation.fromstring("complement(3..2)", 4, True)),
-            "join{[0:2](-), [2:4](-)}",
-        )
+            # Origin-spanning location containing the entire sequence
+            self.assertEqual(
+                str(SimpleLocation.fromstring("3..2", 4, True)), "join{[2:4], [0:2]}"
+            )
+            self.assertEqual(
+                str(SimpleLocation.fromstring("complement(3..2)", 4, True)),
+                "join{[0:2](-), [2:4](-)}",
+            )
 
 
 if __name__ == "__main__":

